### PR TITLE
Add guide for uploading custom images with imgur

### PR DESCRIPTION
**Pull request by paked**

### CODE REVIEW

**Comment by saescapa** [comment]
And maybe some alternative site if they dislike imgur (I know some people do because it is completely public) like imageshack i think is one. And also if it's not there, local upload if they are using c9.

**Comment by saescapa** [comment]
What I meant with public is that anyone can easily find the picture, I would need to do more research on that tho.

**Comment by paked** [comment]
@saescapa I think adding a note that what you upload here will be public and giving a few links to alternative services should be fine

**Comment by zachlatta** [comment]
Won't it be public on Imageshack as well? Any site where you can link
directly to an image needs to make the image public.

**Comment by MaxWofford** [comment]
Even hosting on c9 will technically be public.

**Comment by zachlatta** [comment]
I guess the question is whether Imgur lists all images that are uploaded in
their image index.

On Thu, Dec 31, 2015 at 4:15 PM, Max Wofford <notifications@github.com>
wrote:

> Even hosting on c9 will technically be public.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/615#issuecomment-168265731>.
>



-- 
Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

**Comment by MaxWofford** [comment]
Imgur's help pages are down right now, but my understanding is that it only shows up in 'User Sub' if you are logged in (anonoymous uploads are unlisted).

**Comment by jonleung** [comment]
@MaxWofford it would be good for us to double check that. I personally think that using Imgur is fine.

**Comment by jonleung** [comment]
@paked and I were talking and it seems that because they're eventually going to put the images on a public facing website anyway, it's probably fine to have to have their images listed in Imgur. I want to go ahead and just merge this in because I know that people _really_ wanted to have these directions during Hack Series. Are people comfortable with this?

**Comment by zachlatta** [comment]
Yeah, I'm cool with it.

On Fri, Jan 1, 2016 at 3:06 PM, Jonathan Leung <notifications@github.com>
wrote:

> @paked <https://github.com/paked> and I were talking and it seems that
> because they're eventually going to put the images on a public facing
> website anyway, it's probably fine to have to have their images listed in
> Imgur. I want to go ahead and just merge this in because I know that people
> *really* wanted to have these directions during Hack Series. Are people
> comfortable with this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/615#issuecomment-168348048>.
>



-- 
Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

File: workshops/portfolio/IMGUR.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+![correct link](img/correct_link.gif)
+
+## Making it Appear on Your Page
+
+Once you have this link you can use it in an `<img>` tag just like any other
+image. Here is an example, as a reminder.
+
+```html
+<img src="http://i.imgur.com/3QCaP91.png">
+```
```
Comment (zachlatta) [inline workshops/portfolio/IMGUR.md]: Why is this file's name capitalized?

**Comment by saescapa** [comment]
Sounds good to me 👍

File: workshops/portfolio/IMGUR.md (hunk 1 above)
Comment (jonleung) [inline workshops/portfolio/IMGUR.md]: @paked?

**Comment by jonleung** [comment]
Assigning back to @paked to address this comment: https://github.com/hackclub/hackclub/pull/615#discussion_r48681026

**Comment by paked** [comment]
@zachlatta This PR is ready to be merged, but markcop seems to be throwing a fit about the reference links in the Twilio workshops again?

**Comment by MaxWofford** [comment]
@paked This should be resolved by https://github.com/hackclub/hackclub/pull/645

**Comment by MaxWofford** [comment]
@paked #645 was merged. You  can update your branch and the tests will pass.

**Comment by MaxWofford** [comment]
Just curious, have people reached out saying they've had issues with uploading to imgur?

**Comment by zachlatta** [comment]
I haven't heard of anyone specifically reaching out about it, but when
we've run the workshop in person, one of the most common questions is how
to add a custom image to the website.

One potential problem with this pull request is that Imgur is blocked on
nearly every school network.

On Tue, Jan 12, 2016 at 6:17 AM, Max Wofford <notifications@github.com>
wrote:

> Just curious, have people reached out saying they've had issues with
> uploading to imgur?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/615#issuecomment-170924320>.
>



-- 
Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

**Comment by MaxWofford** [comment]
Let's change references of imgur to surrogate

**Comment by zachlatta** [comment]
I don't think the uploading works with surrogate because it appends /
imgur.com/ to the path (see https://surrogate.hackedu.us/imgur.com/).

On Tue, Jan 12, 2016 at 12:54 PM, Max Wofford <notifications@github.com>
wrote:

> Let's change references of imgur to surrogate
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/615#issuecomment-171051599>.
>



-- 
Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

**Comment by zachlatta** [comment]
I hate to do this, but I'm going to go ahead and close this pull request because @jonleung is rewriting the Portfolio workshop with the updated tools (Cloud9 and GitHub Pages).

@jonleung can you include instructions for uploading custom images to the site in the new portfolio workshop?

Thank you for this @paked though. Truly.

**Comment by paked** [comment]
No problem. I'm happy to have brought up a relevant point of discussion.

**Comment by MaxWofford** [comment]
@zachlatta do you think a surrogate imgur would be useful (single page that you can upload an image through and it spits out a surrogate link)?

**Comment by jonleung** [comment]
> @zachlatta do you think a surrogate imgur would be useful (single page that you can upload an image through and it spits out a surrogate link)?

@MaxWofford, I started to write https://github.com/hackclub/imgur-uploader-surrogate intending it to be a "surrogate imgur". However it is not finished yet. This line, https://github.com/hackclub/imgur-uploader-surrogate/blob/master/js/imgur.js#L20, needs to be rewritten to use surrogate.

**Comment by MaxWofford** [comment]
What if you left it as is and just added the string 'surrogate.hackclub.io/' to the url it returns?

**Comment by jonleung** [comment]
@MaxWofford, would the new URL look like this?

`https://surrogate.hackclub.io/api.imgur.com/2/upload.json`

It didn't seem to work:
> image

/cc @zachlatta

**Comment by zachlatta** [comment]
https://surrogate.hackedu.us (not https://surrogate.hackclub.io)

Also, is there another image uploading service we might be able to use?
It's a huge pain to have to proxy everything and I can totally see Imgur
blocking us for doing this.

On Tue, Jan 12, 2016 at 3:56 PM, Jonathan Leung <notifications@github.com>
wrote:

> @MaxWofford <https://github.com/MaxWofford>, would the new URL look like
> this?
>
> https://surrogate.hackclub.io/api.imgur.com/2/upload.json
>
> It didn't seem to work:
>
> [image: image]
> <https://cloud.githubusercontent.com/assets/654941/12280716/fb9f6a6a-b944-11e5-8e8f-fecc8d580049.png>
>
> /cc @zachlatta <https://github.com/zachlatta>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/615#issuecomment-171103787>.
>



-- 
Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

**Comment by jonleung** [comment]
@zachlatta, many of the services that I have looked through seem to be blocked...

**Comment by MaxWofford** [comment]
Hackers can self host with c9

**Comment by zachlatta** [comment]
OH YEAH. Totally forgot that we're _actually using a real development environment now_ :smile:.

And it'll work on GitHub Pages too.

**Comment by jonleung** [comment]
Oh yeah lol

**Comment by jonleung** [comment]
LOL

**Pull request closed without merging**